Proteomics/Protein/Protein.cs

```diff
@@ -820,6 +820,46 @@ private void AddIntactProteoformToProteolysisProducts(InitiatorMethionineBehavio
             }
         }
 
+        public void CleaveOnceBetweenProteolysisProducts()
+        {
+            List<int> cleavagePostions = new();
+            List<int> proteolysisProductEndPositions = _proteolysisProducts.Where(p => p.OneBasedEndPosition.HasValue).Select(p => p.OneBasedEndPosition.Value).ToList();
+
+            if(proteolysisProductEndPositions.Count > 0)
+            {
+                foreach (int proteolysisProductEndPosition in proteolysisProductEndPositions)
+                {
+                    if(_proteolysisProducts.Any(p=>p.OneBasedBeginPosition == (proteolysisProductEndPosition + 1)))
+                    {
+                        cleavagePostions.Add(proteolysisProductEndPosition);
```

Comment:
Nothing added to cleavagePositions list if proteolysisProductEndPositions is greater than zero. So the foreach loop on line 839 will still evaluate the empty list.